mypy/junit.py

```diff
@@ -92,19 +210,52 @@ def create_serious_error_document(
     sources: List[BuildSource],
     messages: List[str],
 ) -> JUnitXMLDocument:
-    total_elapsed_secs = (finished_at - started_at).total_seconds()
+    total_elapsed_time = finished_at - started_at
 
     # If there was a serious error, we consider the test suite to be one case that failed
-    error_case = TestCase('mypy', classname='mypy', elapsed_sec=total_elapsed_secs)
-    error_case.add_error_info(
+    error_case = TestCase(
+        'mypy',
+        classname='mypy',
+        elapsed_time=total_elapsed_time,
+        result_state=TestCaseResultState.error,
         message="mypy produced messages",
         output=_format_messages_for_test_case(messages),
     )
-
     test_suite = TestSuite(
         'mypy',
-        timestamp=started_at.isoformat(),
+        timestamp=started_at,
         test_cases=[error_case],
     )
 
     return JUnitXMLDocument(test_suite)
+
+
+def _clean_illegal_xml_chars(string_to_clean: str) -> str:
+    """
+    Removes any illegal unicode characters from the given XML string.
+    Based on: https://stackoverflow.com/q/1707890
+    """
+    illegal_unichrs = [
+        (0x00, 0x08), (0x0B, 0x0C), (0x0E, 0x1F),
+        (0x7F, 0x84), (0x86, 0x9F), (0xFDD0, 0xFDDF),
+        (0xFFFE, 0xFFFF),
+    ]
+    if sys.maxunicode >= 0x10000:
+        illegal_unichrs.extend([
+            (0x1FFFE, 0x1FFFF), (0x2FFFE, 0x2FFFF),
+            (0x3FFFE, 0x3FFFF), (0x4FFFE, 0x4FFFF),
+            (0x5FFFE, 0x5FFFF), (0x6FFFE, 0x6FFFF),
+            (0x7FFFE, 0x7FFFF), (0x8FFFE, 0x8FFFF),
+            (0x9FFFE, 0x9FFFF), (0xAFFFE, 0xAFFFF),
+            (0xBFFFE, 0xBFFFF), (0xCFFFE, 0xCFFFF),
+            (0xDFFFE, 0xDFFFF), (0xEFFFE, 0xEFFFF),
+            (0xFFFFE, 0xFFFFF), (0x10FFFE, 0x10FFFF),
+        ])
+
+    illegal_ranges = [
+        "%s-%s" % (chr(low), chr(high))
```

Comment:
This would seem to include a null byte in the regexp (since the very first range starts with 0x00). That seems asking for trouble, even if it works. I recommend putting \x or \u escapes in the regexp instead.